types/state/stateTxn.go

```diff
@@ -0,0 +1,427 @@
+package state
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"github.com/AccumulateNetwork/accumulate/internal/logging"
+	"github.com/AccumulateNetwork/accumulate/smt/common"
+	"github.com/AccumulateNetwork/accumulate/smt/managed"
+	"github.com/AccumulateNetwork/accumulate/smt/pmt"
+	"github.com/AccumulateNetwork/accumulate/smt/storage"
+	"github.com/AccumulateNetwork/accumulate/smt/storage/database"
+	"github.com/AccumulateNetwork/accumulate/types"
+	"github.com/tendermint/tendermint/libs/log"
+	"sort"
+	"sync"
+	"time"
+)
+
+type DBTransactional struct {
+	db           *database.Manager
+	mm           *managed.MerkleManager
+	debug        bool
+	bpt          *pmt.Manager //pbt is the global patricia trie for the application
+	blockIndex   int64        //Index of the current block
+	TimeBucket   float64
+	mutex        sync.Mutex
+	updates      map[types.Bytes32]*blockUpdates
+	writes       map[storage.Key][]byte
+	transactions transactionLists
+	sync         sync.WaitGroup
+	logger       log.Logger
```

Comment:
Instead of copying all the fields of `StateDB`, `DbTransactional` should have a `db *StateDB` field and should access other fields through that one.